src/v/cluster/rm_stm.cc

```diff
@@ -1152,12 +1152,11 @@ void rm_stm::reset_seq(model::batch_identity bid, model::term_id term) {
     seq_it.entry.last_write_timestamp = model::timestamp::now().value();
 }
 
-ss::future<result<kafka_result>> rm_stm::transactional_replicate(
-  model::batch_identity bid, model::record_batch_reader br) {
-    if (!check_tx_permitted()) {
-        co_return errc::generic_tx_error;
-    }
-
+ss::future<result<kafka_result>> rm_stm::do_sync_and_transactional_replicate(
```

Comment:
the commit message looks like this change is pure refactoring but actually it changes tx logic to dedupe and wait on in-flight requests